multinode-demo/gce_multinode.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+command=$1
+prefix=
+num_nodes=
+out_file=
+image_name="ubuntu-16-04-cuda-9-2-new"
+
+OPTIND=$OPTIND+1
+
+usage() {
+  exitcode=0
+  if [[ -n "$1" ]]; then
+    exitcode=1
+    echo "Error: $*"
+  fi
+  cat <<EOF
+usage: $0 <create/delete> <-p prefix> <-n num_nodes> <-o file> [-i image-name]
+
+Creates a GCE multinode network
+
+ create/delete    - Create or delete the network
+ -p prefix        - A common prefix for node names, to avoid collision
+ -n num_nodes     - Number of nodes
+ -o out_file      - Used for create option. Outputs an array of IP addresses
+                    of new nodes to the file
+ -i image_name    - Existing image on GCE (default $image_name)
+
+EOF
+  exit $exitcode
+}
+
+while getopts "h?p:i:n:o:" opt; do
+  case $opt in
+  h | \?)
+    usage
+    exit 0
+    ;;
+  p)
+    prefix="$OPTARG"
+    ;;
+  i)
+    image_name="$OPTARG"
+    ;;
+  o)
+    out_file="$OPTARG"
+    ;;
+  n)
+    num_nodes="$OPTARG"
+    ;;
+  *)
+    usage "Error: unhandled option: $opt"
+    ;;
+  esac
+done
+
+set -e
+
+if [[ -z "$command" ]]; then
+  usage
+fi
+
+if [[ -z "$prefix" ]]; then
+  usage
+fi
+
+if [[ -z "$num_nodes" ]]; then
+  usage
+fi
+
+nodes=()
+for i in $(seq 1 "$num_nodes"); do
+  nodes+=("$prefix$i")
+done
+
+if [ "$command" = "create" ]; then
+  if [[ -z "$out_file" ]]; then
+    usage
+  fi
+
+  ip_addr_list=$(gcloud beta compute instances create "${nodes[@]}" --zone=us-west1-b --tags=testnet \
```

Comment:
you could instead do:
```
ip_addr_array=( $(gcloud beta compute instances create "${nodes[@]}" --zone=us-west1 ... )
declare -p ip_addr_array > "$out_file"
```

Reply:
This change is giving me shellcheck errors.

Reply:
directive it away, shellcheck is sometimes dumb